lib/custom/src/MShop/Service/Provider/Payment/PaypalPlus.php

```diff
@@ -0,0 +1,303 @@
+<?php
+
+/**
+ * @license LGPLv3, http://opensource.org/licenses/LGPL-3.0
+ * @copyright Aimeos (aimeos.org), 2015-2020
+ * @package MShop
+ * @subpackage Service
+ */
+
+
+namespace Aimeos\MShop\Service\Provider\Payment;
+
+
+use Omnipay\Omnipay as OPay;
+use Aimeos\MShop\Order\Item\Base as Status;
+
+/**
+ * Payment provider for payment gateways supported by the PaypalPlus library.
+ *
+ * @package MShop
+ * @subpackage Service
+ */
+class PaypalPlus
+extends    \Aimeos\MShop\Service\Provider\Payment\OmniPay
+	implements \Aimeos\MShop\Service\Provider\Payment\Iface
+{
+	private $beConfig = array(
+		'clientid' => array(
+			'code' => 'clientid',
+			'internalcode'=> 'clientid',
+			'label'=> 'client id from paypal',
+			'type'=> 'string',
+			'internaltype'=> 'string',
+			'default'=> '0',
+			'required'=> true,
+		),
+		'secret' => array(
+			'code' => 'secret',
+			'internalcode'=> 'secret',
+			'label'=> 'secret string from paypal',
+			'type'=> 'string',
+			'internaltype'=> 'string',
+			'default'=> '0',
+			'required'=> true,
+		),
+		'onsite' => array(
+			'code' => 'onsite',
+			'internalcode'=> 'onsite',
+			'label'=> 'Collect data locally',
+			'type'=> 'boolean',
+			'internaltype'=> 'boolean',
+			'default'=> '0',
+			'required'=> true,
+		),
+		'cancelUrl' => array(
+			'code' => 'cancelurl',
+			'internalcode'=> 'cancelurl',
+			'label'=> 'cancelurl for paypal',
+			'type'=> 'string',
+			'internaltype'=> 'string',
+			'default'=> 'www.example.com',
+			'required'=> false,
+		),
+		'address' => array(
+			'code' => 'address',
+			'internalcode'=> 'address',
+			'label'=> 'Send address to payment gateway too',
+			'type'=> 'boolean',
+			'internaltype'=> 'boolean',
+			'default'=> '0',
+			'required'=> false,
+		),
+		'authorize' => array(
+			'code' => 'authorize',
+			'internalcode'=> 'authorize',
+			'label'=> 'Authorize payments and capture later',
+			'type'=> 'boolean',
+			'internaltype'=> 'boolean',
+			'default'=> '1',
+			'required'=> true,
+		),
+		'createtoken' => array(
+			'code' => 'createtoken',
+			'internalcode'=> 'createtoken',
+			'label'=> 'Request token for recurring payments',
+			'type'=> 'boolean',
+			'internaltype'=> 'boolean',
+			'default'=> '1',
+			'required'=> true,
+		),
+		'testmode' => array(
+			'code' => 'testmode',
+			'internalcode'=> 'testmode',
+			'label'=> 'Test mode without payments',
+			'type'=> 'boolean',
+			'internaltype'=> 'boolean',
+			'default'=> '0',
+			'required'=> true,
+		),
+	);
+	private $provider;
+
+	private $feConfig = array(
```

Comment:
Don't need this at all but we need the extension of the checkout process template